File: Company.Registry.Client/src/types/Company.ts
export interface Company {
  id: string;
  name: string;
  exchange: string;
  ticker: string;
  isin: string;
  website?: string;
}

export interface CreateCompanyRequest {
  name: string;
  exchange: string;
  ticker: string;
  isin: string;
  website?: string;
}

export interface UpdateCompanyRequest {
  id: string;
  name: string;
  exchange: string;
  ticker: string;
  isin: string;
  website?: string;
} 

File: Company.Registry/Domain/Entities/Company.cs
namespace Domain.Entities;

public class Company
{
    public Guid Id { get; init; }

    public required string Name { get; set; }

    public required string Exchange { get; set; }

    public required string Ticker { get; set; }

    public required string Isin { get; set; }

    public string? Website { get; set; }
}